File: Callum/day3_1.java
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.util.List;

public class day3_1 {

    public static long getJoltage() throws IOException {
        File inputFile = new File("Callum/inputs/day3_1.txt");
        List<String> lines = Files.lines(inputFile.toPath()).toList();
        long result = 0;

        for (String line : lines) {
            String firstDigit = findBiggestDigit(line.substring(0, line.length()-1));
            String secondDigit = findBiggestDigit(line.substring(line.indexOf(firstDigit)+1));
            String number = firstDigit + secondDigit;
            result += Long.parseLong(number);
        }
        return result;
    }

    private static String findBiggestDigit(String substring) {
        return String.valueOf((char) substring.chars().max().getAsInt());
    }
}

File: rathias/day3_1.lua
local path = "../inputs/day3_1.txt"

local totalJoltage = 0
for line in io.lines(path) do
	local firstDigit, secondDigit = 0, 0
	for i = 9, 1, -1 do
		local firstIndex = string.find(line:sub(1, #line - 1), tostring(i))
		if firstIndex then
			firstDigit = i
			for j = 9, 1, -1 do
				local secondIndex = string.find(line:sub(firstIndex + 1), tostring(j))
				if secondIndex then
					secondDigit = j
					break
				end
			end
			break
		end
	end
	totalJoltage = totalJoltage + tonumber(firstDigit .. secondDigit)
	-- print(firstDigit .. " " .. secondDigit)
end
print(totalJoltage)
